source/common/upstream/host_utility.cc

```diff
@@ -196,5 +197,71 @@ HostConstSharedPtr HostUtility::selectOverrideHost(const HostMap* host_map, Host
   return nullptr;
 }
 
+void HostUtility::forEachHostMetric(
+    const ClusterManager& cluster_manager,
+    const std::function<void(Stats::PrimitiveCounterSnapshot&& metric)>& counter_cb,
+    const std::function<void(Stats::PrimitiveGaugeSnapshot&& metric)>& gauge_cb) {
+  for (const auto& [unused_name, cluster_ref] : cluster_manager.clusters().active_clusters_) {
+    if (cluster_ref.get().info()->perEndpointStatsEnabled()) {
+      const std::string cluster_name =
+          Stats::Utility::sanitizeStatsName(cluster_ref.get().info()->observabilityName());
+
+      const Stats::TagVector& fixed_tags =
+          cluster_ref.get().info()->statsScope().store().fixedTags();
+
+      for (auto& host_set : cluster_ref.get().prioritySet().hostSetsPerPriority()) {
+        for (auto& host : host_set->hosts()) {
+
+          Stats::TagVector tags;
```

Comment:
perf nit: hoist the tags declaration out of these two loops, below where fixed_tags is declared. We only need to reset the host tag, not re-declare the vector and re-copy the cluster_name and fixed tags.

Reply:
I'll write the benchmark you mention below, get a baseline with current version, then play around and see if this makes it meaingfully faster.

I think we could remove one allocation without much extra complexity by hoisting the declaration out of loops and calling `clear()` it right here everytime, so the backing array doesn't get reallocated everytime.

Reply:
the tag for the cluster name also will be invariant, right? That would save more string-copies, right?

Reply:
Ok, benchmark program is working. There was zero performance change by hoisting as much as possible out of the loop.